packages/block-library/src/post-comments/index.php

```diff
@@ -15,17 +15,20 @@ function render_block_core_post_comments() {
 	if ( ! $post ) {
 		return '';
 	}
-	$comments = get_comments(
-		array(
-			'post_id' => $post->ID,
-		)
-	);
-	$output   = '';
-	// TODO: Handle nested comments.
-	foreach ( $comments as $comment ) {
-		$output .= '<p>' . $comment->comment_author . '<br />' . $comment->comment_content . '</p>';
-	}
-	return $output;
+
+	ob_start();
+	// This generates a deprecate message.
+	// Ideally this deprecation is removed.
+	comments_template( ABSPATH . WPINC . '/theme-compat/comments.php' );
+	ob_get_clean();
+
+	ob_start();
+	wp_list_comments(array(
```

Comment:
I don't think we need to call `wp_list_comments` explicitly here since it's already called in `comments.php`. If I'm not missing something, it seems that this can be simplified to:

```php
ob_start();

comments_template();

return ob_get_clean();
```

We can optionally call it with `dummy_path` or similar if we want to force the usage of theme-compat template.

Reply:
We can't do that since this function generates the deprecated message.
Also, it seems that this function doesn't output the HTML consistently.

> Will not display the comments template if not on single post or page, or if the post does not have comments.

https://developer.wordpress.org/reference/functions/comments_template/